server/src/routes/upload/parseBeatmap.ts

```diff
@@ -101,6 +101,26 @@ export const parseBeatmap: (
     hash.update(b)
   }
 
+  const difficultiesData = await Promise.all(
+    [1, 3, 5, 7, 9].map(async rank => {
+      const diff = difficulties.find(x => x._difficultyRank === rank)
+      if (!diff) {
+        return null
+      }
```

Comment:
Lines 107-109 could be simplified

```suggestion
      if (!diff) return null
```